tf_encrypted/layers/dense.py

```diff
@@ -37,25 +40,39 @@ def initialize(
         if initial_weights is None:
             initial_size = (self.in_features, self.out_features)
             initial_weights = np.random.normal(scale=0.1, size=initial_size)
-        if initial_bias is None:
-            initial_bias = np.zeros((1, self.out_features))
+        if initial_bias is not None:
+            self.bias = self.prot.define_private_variable(initial_bias)
 
         self.weights = self.prot.define_private_variable(initial_weights)
-        self.bias = self.prot.define_private_variable(initial_bias)
 
     def forward(self, x):
         self.layer_input = x
-        y = x.matmul(self.weights) + self.bias
+
+        if self.transpose_input:
+            self.layer_input = self.layer_input.transpose()
```

Comment:
would this transpose more than once if `forward` is called repeatedly?

Reply:
I think we should be good because every time we run forward, self.layer_input will be assign first to `x`, which is the forward input. However, we will have an issue for the weight matrix. I think we could move this weight transpose to the initialize phase.